Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt

```diff
@@ -176,59 +175,84 @@ fun DetailsContent(
     }
 }
 
+/**
+ * CityMapView
+ * A composable that shows a map centered on a location with a marker.
+ */
 @Composable
-private fun CityMapView(latitude: String, longitude: String) {
-    // The MapView lifecycle is handled by this composable. As the MapView also needs to be updated
-    // with input from Compose UI, those updates are encapsulated into the MapViewContainer
-    // composable. In this way, when an update to the MapView happens, this composable won't
-    // recompose and the MapView won't need to be recreated.
-    val mapView = rememberMapViewWithLifecycle()
-    MapViewContainer(mapView, latitude, longitude)
-}
-
-@Composable
-private fun MapViewContainer(
-    map: MapView,
+fun CityMapView(
     latitude: String,
-    longitude: String
+    longitude: String,
+    cameraPositionState: CameraPositionState = rememberCameraPositionState(),
+    onMapLoaded: () -> Unit = {},
 ) {
-    val cameraPosition = remember(latitude, longitude) {
+    val cityLocation = remember(latitude, longitude) {
         LatLng(latitude.toDouble(), longitude.toDouble())
     }
 
-    LaunchedEffect(map) {
-        val googleMap = map.awaitMap()
-        googleMap.addMarker { position(cameraPosition) }
-        googleMap.moveCamera(CameraUpdateFactory.newLatLng(cameraPosition))
+    // Set our initial camera position whenever the city location changes.
+    LaunchedEffect(cityLocation) {
```

Comment:
```suggestion
    LaunchedEffect(cityLocation, cameraPositionState) {
```

As `cameraPositionState` is also used in the effect block, we should "key" it as well

Reply:
That would break it, it's only meant to happen on first load. I think I am just going to fully remove it and pass in only a cameraPositionState. There is definitely something not right about this section